### PR TITLE
Feat/favs

### PR DESCRIPTION
**Pull request by harrietmorris**

### CODE REVIEW

File: client/context/globalContext.tsx [hunk 1]
```diff
@@ -9,15 +9,12 @@ const initialDataContext: DataContextProps = {
   user: null,
   setUser: () => {},
   profiles: [],
-  setProfiles: () => [],
+  setProfiles: () => {},
```
Comment (sebastiandve) [inline client/context/globalContext.tsx:12]: this should be a list, no?

File: client/context/globalContext.tsx [hunk 2]
```diff
@@ -40,12 +36,11 @@ export const GetUserProvider = ({ children }: ProviderProps) => {
         setProfiles,
         selectedProfile,
         setSelectedProfile,
-        allStories,
-        setAllStories,
+        // allStories,
+        // setAllStories,
```
Comment (sebastiandve) [inline client/context/globalContext.tsx:40]: remove comments

File: client/context/globalContext.tsx [hunk 3]
```diff
@@ -40,12 +36,11 @@ export const GetUserProvider = ({ children }: ProviderProps) => {
...
         selectedStory,
         setSelectedStory,
-        favStories,
-        setFavStories,
+
```
Comment (sebastiandve) [inline client/context/globalContext.tsx:43]: remove line

File: client/services/apiService.ts [hunk 4]
```diff
@@ -109,4 +96,25 @@ export const newProfile  = async (userId: number, profileData: ProfileType): Pro
...
     throw e;
   }
-}
\ No newline at end of file
+}
+
+export const addToFavs = async (profileId: number, storyId: number): Promise<ProfileType> => {
+  try {
+    const response: AxiosResponse<ProfileType> = await axios.put(`${BASE_URL}/profiles/${profileId}/favs/${storyId}`);
+    // console.log('adding to favs', response.data);
```
Comment (sebastiandve) [inline client/services/apiService.ts:104]: comment

File: client/components/buttons/favButton.tsx [hunk 5]
```diff
@@ -1,38 +1,55 @@
...
-            : [...selectedProfile.favs, { id: storyId } as StoryType]; 
+    // const isFav = selectedProfile.favs.some(fav => fav.id === storyId);
+
+    // const toggleFavorite = () => {
+    //     const updatedFav = isFav
+    //         ? selectedProfile.favs.filter(story => story.id !== storyId)
+    //         : [...selectedProfile.favs, { id: storyId } as StoryType]; 
+
+    //     setSelectedProfile({ ...selectedProfile, favs: updatedFav });
+    // };
```
Comment (sebastiandve) [inline client/components/buttons/favButton.tsx:46]: get rid of this?

**Review by sebastiandve — CHANGES_REQUESTED** [review]
some minor details

File: client/components/buttons/favButton.tsx [hunk 6]
```diff
@@ -1,38 +1,55 @@
...
 
-    const toggleFavorite = () => {
-        const updatedFav = isFav
-            ? selectedProfile.favs.filter(story => story.id !== storyId)
-            : [...selectedProfile.favs, { id: storyId } as StoryType]; 
+    // const isFav = selectedProfile.favs.some(fav => fav.id === storyId);
+
+    // const toggleFavorite = () => {
+    //     const updatedFav = isFav
+    //         ? selectedProfile.favs.filter(story => story.id !== storyId)
```
Comment (Badrhoul) [inline client/components/buttons/favButton.tsx:42]: do we keep or delete this comment?

File: client/context/globalContext.tsx (hunk 1 above)
Comment (harrietmorris) [reply client/context/globalContext.tsx:12]: its a function : 
  profiles: ProfileType[];
  setProfiles: React.Dispatch<React.SetStateAction<ProfileType[]>>;

**Review by sebastiandve — APPROVED** [review]
lgtm

**Pull request merged**